[gdal] Add Static Library Dependencies

n4z4m3 opened the pull request:
Add geos.lib, shell32.lib, and ole32.lib so that downstream projects do not need to explicitly include them

n4z4m3 committed:
Add Static Library Dependencies

Add geos.lib, shell32.lib, and ole32.lib so that downstream projects do not need to explicitly include them

n4z4m3 committed:
Add Static Library Dependencies

Add geos.lib, shell32.lib, and ole32.lib so that downstream projects do not need to explicitly include them

nazame commented:
If somebody could figure out why this is failing the x86_windows check that would be great. It is not failing for me when I build locally. Could it just be timing out? Because it was failing on the x64_windows and x64_windows static before I updated the CONTROL file and it re-did the checks. So I don't think there is a real problem.

I clicked on through the error details and it gets to this:

_Found the following errors:
Error: While reading versions for port gdal from file: C:\a\1\s\versions\g-\gdal.json
       Version `3.1.3#3` was not found in versions file.
       Run:
           vcpkg x-add-version gdal
       to add the new port version._

What does that mean? Is that x-add-version something I am supposed to do?

Thanks in advance.

NancyLi1013 commented:
Hi @nazame 

Thanks for your PR.

For the failure on x86-windows, you can run `.\vcpkg x-add-version gdal
` and then submit the changes for versions to solve.

nazame commented:
Thank you @NancyLi1013 

Is that x-add-versions a new command that is required for contributing changes? 
I guess I need to remember that one for next time.

NancyLi1013 commented:
Yes, it is a new one. You need to run it every time when contributing.

Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-files-in-versions-of-any-modified-ports

https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.reference.md#updating-the-version-files

nazame commented:
I just realized that this also resolves issue #15526

nazame commented:
Oops. I accidentally clicked the close button. How do I unclose this?

JackBoosY commented on ports/gdal/portfile.cmake:
```diff
@@ -88,7 +88,7 @@ if (VCPKG_TARGET_IS_WINDOWS)
       GDAL_HOME=${CURRENT_PACKAGES_DIR}
       CXX_CRT_FLAGS=${LINKAGE_FLAGS}
-      PROJ_LIBRARY=${PROJ_LIBRARY_REL}
+      "PROJ_LIBRARY=${PROJ_LIBRARY_REL} shell32.lib ole32.lib"
```
Can you double confirm this?
What happens when you don’t add it?

nazame replied:
If I don't add this, then my downstream projects require me to add it. This was the main reason for this pull request. That and the geos.lib

JackBoosY replied:
Can you please provide the failure logs?

nazame replied:
I am sorry. I may be causing confusion by using the incorrect terminology. When I say I see the error in "downstream" projects what I mean are in projects that use the gdal library. There are no failures in compiling the gdal vcpkg package or any other package.

The problem is due to some change in the proj library (see this https://github.com/OSGeo/gdal/issues/2743)

When I compile my project with Visual Studio 2019 in Windows and I do not include ole32.lib as an Additional Dependency in the linker settings, I get the following error.

LNK2001	unresolved external symbol __imp_CoTaskMemFree	

This started happening with the previous updates to the vcpkg library when it went from 2.4.x to 3.1.x which also updated its required version of the proj library.

Note that I am not including  shell32.lib here because I have a lot of projects that use gdal and I am sure that I was getting a different linkage error with that missing as well, but I can't find the couple projects that had that error, But maybe it was actually due to some other feature I was using in that project that required it, so maybe we can remove shell32.lib from my change here. I can always add it back later if I find that it is indeed needed for some functionality within gdal/proj.

Also note that technically you can use the option in Visual Studio to "inherit from parent or project defaults" for the Additional Dependencies which will include ole32.lib and shell32.lib but I figure that not everybody using the gdal library will know to do this even though it is the default setting. Instead they may face this linker error and may have a difficult time figuring out how to resolve it.

I also figured that a few lines down from this one in the portfile.cmake shell32.lib is already included with PG_LIB (which is another reason why I am second guessing using shell32.lib here)

So in conclusion, I would be OK removing shell32.lib, but I really do believe that ole32.lib is helpful for others who may not know how to resolve that linker issue.

JackBoosY replied:
![image](https://user-images.githubusercontent.com/47264268/109919025-ef151c00-7cf2-11eb-8f21-6302c2681243.png)
Seems these 2 system libraries are not required.

nazame replied:
Actually, I forgot to mention that this is for building in static mode so you are right that the DLL does not have the dependency.

And it is definitely a dependency in the static version although it is really a dependency of a dependency (proj.lib) so maybe the gdal vcpkg package is the wrong place to implement a fix since it is possibly more of a workaround. Maybe the proj library is the proper place.

I know it is a dependency because the only thing that changed for my projec to make the issue appear was when I updated my gdal vcpkg which is a dependency of my project (links with gdal.lib statically) and did a recompile. That is when the linking error appeared and I tracked it down to ole32.lib.

I am fine with leaving this change out because as I said I already know how to resolve it in my project by linking with ole32.lib, but I figured it might be helpful to others.

So I will make the change to remove these .lib dependencies

JackBoosY replied:
If this only happens on static build. I think that's okay.

JackBoosY commented:
And please notice #16088 and #15808 makes almost same changes.

vicroms commented:
/azp run

JackBoosY commented:
Can you please solve the file conflicts?

Thanks.

nazame commented:
> Can you please solve the file conflicts?
> 
> Thanks.

@JackBoosY  I noticed that there have been changes to master since originally forked which are causing the conflicts. The biggest one is that the port has been modified to use the new vcpkg.json instead of CONTROL which is a good thing.
But I am wondering in this case what is the best way to resolve the conflicts. I was thinking that I could re-base. Is that appropriate? Or should I just manually make this branch match the master by deleting the CONTROL, and adding the vcpkg.json and manually making the other changes that are in master?

JackBoosY commented:
@nazame Rebase to master, apply your changes would be okay.

n4z4m3 committed:
Update port version

JackBoosY commented on versions/g-/gdal.json:
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "5d3e68869075ed189fc5b23c2ee85efc4295bd5a",
```
```suggestion
      "git-tree": "1195f30c63703eca670c1675a24dcabf36a5f954",
```

JackBoosY committed:
Update versions/g-/gdal.json

ned14 committed:
[outcome] Replace Outcome single header based port with full fat cmake install port (#15603)

* Replace Outcome single header based port with full fat cmake install port.

* [outcome] Improve portfile.cmake

* update baseline

* Refactor @JackBoosY's changes to make the dependency SHA feature chosen.

* update baseline

* Fix outcome features so they actually work.

* update baseline

* [outcome] install usage

* update baseline

* [outcome] Always use vcpkg copies of dependencies

* DO NOT MERGE YET, ONLY FOR INSPECTION BY PR REVIEW.

1. Break out these dependencies into standalone ports:

  - ned14-internal-quickcpplib
  - status-code

2. Add port for LLFIO.

3. Add dependency smoke tests for Outcome and LLFIO as per instructions.

* Fix all remaining issues, including link failure. This changeset is now ready to merge into vcpkg.

* Update versions.

* Have LLFIO not build static or shared libs according to VCPKG_LIBRARY_LINKAGE.

* Fix last commit breaking everything :)

* Update versions.

* Disable uwp support for LLFIO.

* versions

* Fix status code on GCC 6 and 7

* Don't download ntkernel-error-category if not on Windows.

* Fix status-code on UWP.

* Update versions.

* Fix MSVC on ARM.

* Apply all the feedback from the PR review, more or less.

* versions

* Update to latest, disable UWP for Outcome.

Co-authored-by: JackBoosY <yuzaiyang@beyondsoft.com>
Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

Neumann-A committed:
[qt5-webengine] Fix issues (#16770)

* [qt5-webengine]
 - add feature "proprietary-codecs"
 - fix build error due to internal changes how options are passed.
 - try to fail early on too long build paths instead of late

* comment the ci baseline for x64-windows to actually test it in ci
and only build one configuration to not overstress the ci system

* be less strict about the buildtree path length

* adjust qt5-webengine ci back

* update baseline

* change length error message according to CR.

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* fix version

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

genevanmeter committed:
[ffmpeg] Fix FindFFMPEG lib versions for building opencv[ffmpeg] (#16882)

* Fix FindFFMPEG lib versions for opencv[ffmpeg]

libavutil, libavdevice, etc. define their versions separate from FFMPEG main version in a version.h file located with each component headers.
This change fixes building opencv[ffmpeg] when it attempts to discover the individual versions of FFMPEG libs. Before this change opencv[ffmpeg] would build but the FFMPEG functionality was disabled since all the lib versions were set to 4.3.1

* Modified Port-Version of CONTROL to 13
Ran vcpkg.exe x-add-version ffmpeg

* Ran vcpkg.exe x-add-version ffmpeg --overwrite-version per Cheney-W's instructions

* Updated Port-Version of CONTROL to 1 for FFMPEG version 4.3.2

* Ran vcpkg x-add-version ffmpeg

Neumann-A committed:
[vcpkg/meson] fix some details (#15756)

* [vcpkg_install_meson] add bin to path to run code generators

* [vcpkg_configure_meson]
deactivate native compiler in cross builds
make x86 on x86_x64 a native instead of a cross build
(as long as we are not building for UWP)

* remove double spaces in _FLAGS
remove unnecessary comments

* change regex to take double - into account.

* add missing removal of multiple spaces back into the regex

* remove ws diff change

* remove ws to reduce diff

* Change macro to function

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

* UWP is cross target.

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

Neumann-A committed:
[open62541] add amalgamation feature (#16969)

* [open62541] add amalgamation features

* version stuff

* Update versions/baseline.json

* Update ports/open62541/vcpkg.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* update version file

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

PhoebeHui committed:
[opensubdiv] Remove debug/share folder (#16962)

* [opensubdiv] Remove debug/share folder

* Format the vcpkg.json

* Update baseline revision

Neumann-A committed:
[vcpkg/scripts] add a way to define another tool destination (#16935)

frankhale committed:
[lua]Update port to version 5.4.3 (#16943)

* Update Lua to version 5.4.3

* Update Lua versions file

odygrd committed:
[quill] Upgrade to 1.6.2 (#16931)

* update quill to 1.6.2

* update json file

abique committed:
[fontconfig] fixes for macos (#16905)

References:
 - https://github.com/Homebrew/homebrew-core/pull/57999
 - https://github.com/Homebrew/homebrew-core/issues/57994

Ryan-rsm-McKenzie committed:
[spdlog] Update to 1.8.5 (#16870)

* remove featurebuild patch

see:
https://github.com/gabime/spdlog/commit/3d8f71c4d23fd2df08e4f6284155958b0df47b33
https://github.com/gabime/spdlog/commit/42c36f48ed8d5d64bd244b1097257f575569d472

* remove external fmt patch

see:
https://github.com/gabime/spdlog/blob/53e1c9ab112aa97062ba3df8fede91c3acd248d5/CMakeLists.txt#L183-L184

* remove target fixup branching

see:
https://github.com/gabime/spdlog/commit/4858d7e454e66d347024a074749535b91ccbeed9

* convert control file to manifest file

* remove android build patch

see:
https://github.com/gabime/spdlog/commit/e86f450428fd5373ea72606b21a4055fb411abfd

* update to 1.8.3

* remove redundant cleanup

these folders don't exist in the build tree, and so they don't need to be cleaned up

* update deprecated functions

* remove empty folder

* add version

* report version using semantic versioning

* correct version comment

* update version file

* fixup pkgconfig

* update version file again

* accomodate non-cmake users

* update to 1.8.5

* update version files

* remove 1.8.3 from version file

Thieum committed:
[zstd] update to 1.4.9 (#16865)

* [zstd] update to 1.4.9

Fixes #16642

Update zstd to the latest version 1.4.9: https://github.com/facebook/zstd/releases/tag/v1.4.9

* update baseline / version

* Update vcpkg.json

* Update vcpkg.json

apply suggested diff file

* attempt to fix build errors (missing file libzstd?)

* Update zstd.json

* Update portfile.cmake

* fix version

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

mathisloge committed:
[mapbox-geometry,mapbox-polylabel,geojson-vt-cpp,geojson-cpp] new ports (#16838)

* added geometry

* added mapbox-polylabel

* added versions

* add port geojson-vt-cpp

* add version

* fix formatting

* fix version

* eh ok

* version...

* add geojson

* add version

* false version

* fix internal version

* use semver string

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* fix versions

* added mapbox prefix to name. fixed description

* changed versions

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

JackBoosY committed:
[vcpkg baseline][pangolin] Add options to dependencies to avoid finding them automatically (#17014)

* [vcpkg baseline][pangolin] Add options to dependencies to avoid finding them automatically

* update version record

* switch to CMAKE_DISABLE_FIND_PACKAGE_PORT

* bump version

JonLiu1993 committed:
[podofo] update to 0.9.7 (#16373)

* [podofo] update to 0.9.7

* update version

* [podofo] update to 0.9.7

* update version

* fix-osx-error

* Modify patch format

* update version

* Restore patch endline format

* update version

* update patch file

* update version

* update patch file

* update version

* revert the changes to this file

* update version

* revert the changes to this file

* update version

NickStrupat committed:
[antlr4] update to 4.9.1 (#16383)

* [antlr4] update to 4.9.1

* update version

* [antlr4] fix utfcpp dependency issue

* fix version SHA to account for utfcpp patch

* [antlr4] Minimize patch

Co-authored-by: Jonliu1993 <13720414433@163.com>
Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

NancyLi1013 committed:
[protobuf] Fix deprecation warning in vcpkg_check_feature() (#16997)

* [protobuf] Fix deprecation warning in vcpkg_check_feature()

* Update versions

* Update keyword

* Update versions/p-/protobuf.json

kris-jusiak committed:
[bext-di] Rename `boost-di` to `bext-di` (#16302)

* Rename `boost-di` to `di`

Problem:
- `boost-di` is not an official Boost library and it shouldn't use boost name.

Solution:
- Rename it to `di` instead.

* Update the port name to bext-di

* Update the version

Co-authored-by: Kris Jusiak <kris@jusiak.net>
Co-authored-by: PhoebeHui <20694052+PhoebeHui@users.noreply.github.com>

luncliff committed:
[fp16] create a new port (#16340)

* [psimd] create a new port

* [psimd] update baseline and port SHA

* [fp16] create a new port

* [fp16] update baseline and port SHA

* [fp16] apply PR feedback

luncliff committed:
[pthreadpool] create a new port (#16342)

* [fxdiv] create a new port

* [fxdiv] update baseline and port SHA

* [pthreadpool] create a new port

* [pthreadpool] update baseline and port SHA

* [pthreadpool] update deps for uwp

* [pthreadpool] apply PR feedback

* [pthreadpool] remove unnecessary work in portfile

cpp-niel committed:
[mfl] Add new port (#16345)

* Add control and portfile for mfl port

* Add mfl port

* reference version better header inclusion

* reference version with fmt linkage

* Apply suggestions from code review

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

* Set ports that don't build to fail in the ci baseline

* add version

* Check for static linkage and add message about C++20 requirements.

* add fail to ci baseline for dynamic windows builds

* version update

* Code review improvements

- Improve warning message that the port requires a C++20 compiler
- Use `version` rather than `version-string` in manifest
- Use `"supports": "static"` in manifest instead of setting dynamic
  builds to `fail` in ci baseline.

* update version

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

autoantwort committed:
vcpkg_copy_tool_dependencies: show description how to install powershell (#16347)

JackBoosY committed:
[ceres] Fix static library path on Linux (#15362)

* [ceres] Fix static library path on Linux

* [ceres] Bump version

* update version record

CaeruleusAqua committed:
[sciplot] Add new port (#15913)

* add sciplot port

* port should also work on all platforms, as long as gnuplot can be installed

* add version information to vcpkg

* disable uwp in build pipeline and fix version SHA

* uwp not supported

* fix version sha

* Use vcpkg.json and bump version to 0.2.2

* add sciplot version

* [sciplot] Remove inaccessible 0.1 port

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

coryan committed:
[google-cloud-cpp] update to the latest release (v1.26.0) (#17027)

* [google-cloud-cpp] update to the latest release (v1.26.0)

* Reset port version after update

veggiesaurus committed:
[vtk] enables openvr module when openvr is listed (#17015)

* enables openvr module when openvr is listed

* bumped port number

* add version bump to json files

Commit:
[vcpkg_build_make] pass no parallel command to execute_process when parallel build is disabled (#17018)

Co-authored-by: Ilya Arzhannikov <iarzhannikov@artec3d.com>

skaarj1989 committed:
[entt] Update to 3.7.0 (#17047)

* Update to 3.7.0

* fix

JonLiu1993 committed:
[libmariadb] update to the 3.1.12 (#17004)

* [libmariadb] update to the 3.1.12

* update version

* Add keyword FEATURES

* update version

Neumann-A committed:
[vcpkg/scripts] add variable VCPKG_CROSSCOMPILING (#16900)

* add VCPKG_CROSSCOMPILING

* update docs

* Apply suggestions from code review

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* update docs

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

PhoebeHui committed:
[libarchive] Remove lzo from the default feature list (#16957)

* [libarchive] Remove lzo from the default feature list

* Update the baseline version

JackBoosY committed:
[pthreads] Fix crt linkage (#17011)

* [pthreads] Fix crt linkage

* update version record

* update baseline

mbeutel committed:
[gsl-lite] Update to 0.38.0 (#16984)

* [gsl-lite] Update to 0.38.0

* [gsl-lite] Update version registry

* Re-run "vcpkg x-add-version"

evpobr committed:
[mpg123] Upgrade to 1.26.3-1 (#16910)

* [mpg123] Upgrade to 1.26.3-1

Fix invalid MPG123_API_VERSION value in mpg123.h.in for Windows
platform. It was equal to @API_VERSION@, now it is read from
configure.ac and set to correct value.

See also libsndfile/libsndfile#499.

* [mpg123] Update 1.26.3-1 baseline version

Commit:
[libsoundio] Add support for linux backends (#16891)

* Add support for linux backends

* Run vcpkg format-manifest

* Bump version

* Run vcpkg-format

* Add version

* Use vcpkg_check_features

* Use version instead of version-string

* Update version

* Update versions/l-/libsoundio.json

Co-authored-by: Victor Romero <romerosanchezv@gmail.com>

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>
Co-authored-by: Victor Romero <romerosanchezv@gmail.com>

EwoutH committed:
[dav1d] Update to 0.8.2 (#16916)

* [dav1d] Update to 0.8.2

Update dav1d from 0.8.1 to 0.8.2

* versions: Update dav1d to 0.8.2

* Update version schemes

* Update the baseline version

Co-authored-by: PhoebeHui <20694052+PhoebeHui@users.noreply.github.com>

ras0219-msft committed:
[vcpkg_acquire_msys] Update package versions (#16952)

Neumann-A committed:
[vcpkg/make/scripts]only append the lib directories if they exist. (#16959)

* [vcpkg/make/scripts]only append the lib directories if they exist.

* fix path concat

autoantwort committed:
[libffi] Fix processors (#16285)

* [libffi] fix build on apple silicon

* [libffi] add version files

* [libffi] changes

* [libffi] add version files

* [libffi] Rearrange target conditional list

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

ras0219-msft commented:
This looks like a nice improvement for now, thanks @nazame! Hopefully in the future, we can handle this with pkg-config :)

DigitalInBlue committed:
[celero] Updated to Celero v2.8.0 (#16802)

* [celero] Updated to Celero v2.8.0

* [celero] Improving static linking support in VCPKG.

Celero issue #154 is tracking a fix.
https://github.com/DigitalInBlue/Celero/issues/154

* [celero] Updated to v2.8.1

* [celero] Updated version for VCPKG.

* [celero] Updates to better support VCPKG.

* [celero] Updated versions.

* Update celero.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

Neumann-A committed:
[vcpkg baseline][qt5-base] Fix glib link issues on linux (#16954)

* [qt5-base] fix glib link issues on linux

* add version

* remove unused vars

* update version

* write port info file.

* version stuff

* remove paraview from baseline. It was added due to the same spurious glib failures.

* trying out a different approach

* more changes

* change the if to actually use the buildtype instead

* remove x_vcpkg_get_port_info

* add option QT_OPENSSL_LINK back in

* use INCLUDE_DIRS_(DEBUG|RELEASE) instead of just INCLUDE_DIRS

* regen docs

* remove function call I forgot to remove

* [vcpkg-pkgconfig-get-modules] Move to port

* revert changes to ports.cmake

* include the file in qt5-base

* fix path

* remove unnecessary include

* Apply suggestions from code review

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

* Apply suggestions from code review

* update version

* ws removal

* version stuff

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>
Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

JackBoosY committed:
[vcpkg baseline][gdal] Fix configure error on OSX (#17083)

* [gdal] Fix configure error on OSX

* update version record

* Revert changes

* add touch command

* update version record

Neumann-A committed:
[cmake] update cmake test port (#16948)

* [cmake] update cmake test port

* correct hash
add expat as dependency

* remove double expat dependency

Sibras committed:
[ffmpeg] Add support for dav1d,fontconfig,freetype,fribidi,ilbc,modplug,opengl,openjpeg,libssh,tensorflow,tesseract,webp,libxml2 dependencies. (#15787)

* [tesseract] Use vcpkg_fixup_pkgconfig.

* [libxml2] Correct pkgconfig lib name.

* [libwebp] Use vcpkg_fixup_pkgconfig.

* [libssh] Export pkgconfig on windows.

Also move to using git to get source.

* [modplug] Export pkgconfig on windows.

* [ffmpeg] Add support for fontconfig,freetype,fribidi,modplug,openjpeg,libssh,tesseract,libxml2 dependencies.

* [openjpeg] Correct required static link libs in pkgconfig.

* [modplug] Combine vcpkg_from_github using variable.

* Update ports/libssh/CONTROL

* Improve portfile.cmake

* update version records.

* [openjpeg] Update libs in pkgcfg.

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

* update version records

* [libssh] Add pthread to pkgconfig when using mbedtls.

* [libssh] Correct pthread naming on windows.

* [fontconfig] Add libintl to pkgconfig on windows.

* update version records

* [ffmpeg] Fixup FindFFmpeg.

* [ffmpeg] speex now supports non-windows.

* [ffmpeg] Add feature libass.

* [ffmpeg] Add dav1d feature.

* [ffmpeg] Add feature ilbc.

* [ffmpeg] Add tensorflow feature.

* [ffmpeg] update version record.

* [ffmpeg] Add CI feature test.

* [ffmpeg] Limit features based on CI failures.

* [ffmpeg] Update version record.

* [ffmpeg] limit features based on ci.baseline.

* [various ports] Update supports field.

* [ffmpeg] Limit features more based on CI.

* update version records.

* [ffmpeg] Add detection of additional non target deps..

select_library_configurations_from_names currently detects the debug libs even for release builds as  _IMPORT_PREFIX was not being set.

* [aubio] Silence warning about FindFFMPEG.

* [pangolin] Use vcpkg supplied FindFFMPEG.

* update version files.

* [ffnvcodec] Set as not supporting uwp.

ffnvcodec will build under uwp as its a header only lib, but it can not be used as it requires dynamic dll loading.

* [ffmpeg] Update feature all.

* update version records

* [tesseract] Wrap debug pkgcfg update.

* [libssh,libxml2,openjpeg,fontconfig] Fix pkg-config for release only triplets.

* [libssh] Correct port version after merge.

* [ffmpeg] Fixup after merge.

* Update version files.

* [ffmpeg] Add opengl support.

* [ffmpeg] Update package version.

* [ffmpeg] Fix ffnvcodec support.

* [ffmpeg] Fix x265 detection on osx.

* [libvpx] Enable arm-uwp build.

* [ffmpeg] Fixup x265 patch.

* trigger sdl rebuild

* [ffmpeg] Disable opengl on osx.

* Revert "trigger sdl rebuild"

This reverts commit 94065bfe8414259ad5a5576bd4ac0b7ab4b97c6b.

* [ffmpeg] Disable failing features on osx.

* Update ports/ffmpeg/FindFFMPEG.cmake.in

Co-authored-by: Matthias C. M. Troffaes <matthias.troffaes@gmail.com>

* [ffmpeg] Add ass dependencies to FindFFmpeg.

* Update ports/ffmpeg/FindFFMPEG.cmake.in

Co-authored-by: Matthias C. M. Troffaes <matthias.troffaes@gmail.com>

* update version

* [fontconfig] disable pthread/json as they are not needed for lib builds.

Only used for tests.

* [ffmpeg] Enable fontconfig on static+windows.

* update versions

* update versions.

* Fix incorrectly included commits

* revert pangolin commit

Reverts most of 2543be2edf60bf38511c2d477bb17b617398a108

* update versions

* Correct port version after merge.

* update versions.

* [ffmpeg] Fix cmake dependency detection on non-windows.

* Revert "revert pangolin commit"

This reverts commit f59bc5a53ea74fd9df17fc368886e9d1a80ac42f.

* [ffmpeg] Fix dependency loading that does not define separate debug/release libs.

* update versions.

* [ffmpeg] set CMP0072 policy.

* [wavpack] Fix cmake config export.

* [ffmpeg] Add optional system dependent libraries.

* update versions after merge.

* update versions.

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>
Co-authored-by: JackBoosY <yuzaiyang@beyondsoft.com>
Co-authored-by: Matthias C. M. Troffaes <matthias.troffaes@gmail.com>

autoantwort committed:
format vcpkg.json testing files (#17120)

NancyLi1013 committed:
[libmariadb] Fix build error with cmake 3.20.0 (#17113)

* [libmariadb] Fix build error with cmake 3.20.0

* Update versions

* Add vcpkg.json

* Update versions/l-/libmariadb.json

NancyLi1013 committed:
[fastcdr] Fix support Linux (#17112)

* [fastcdr] Fix support Linux

* Format json and update versions

* Update versions/f-/fastcdr.json

past-due committed:
[sdl2] Fix pthread detection on macOS (#17106)

* [sdl2] Fix pthread detection on macOS

* Add missing "FEATURES" to vcpkg_check_features

* [sdl2] Bump port-version

* Run x-add-version sdl2

huangqinjin committed:
[vcpkg/scripts/make] add compiler tools to PATH (#17085)

* [vcpkg/scripts/make] add compiler tools to PATH

* use find_program to check if tools are already in PATH

SamuelMarks committed:
[nuklear] Upgrade from 2020-09-14 to 2021-03-18 version and switch repo (#17082)

* [nuklear] Upgrade from 2020-09-14 to 2021-03-18 version and switch repo

* Ran `vcpkg x-add-version --all`

Nemirtingas committed:
Update libjuice to 0.7.1 to update libdatachannel (#17076)

* Update libjuice to 0.7.1 to update libdatachannel

* Transform the CONTROL file to .json and update version baseline.

* Update libjuice version. I don't really know why it changed.

* Fix version type

* Fix version type

Co-authored-by: Nemirtingas <nanaki89@hotmail.fr>

autoantwort committed:
[libmt32emu] update to 2.5.0 (#17070)

* [libmt32emu] update to 2.5.0

* add version files

autoantwort committed:
fix VCPKG_CROSSCOMPILING (#17068)

past-due committed:
[sqlite3] Update to 3.35.4 (#17067)

* [sqlite3] Update to 3.35.4

* Run x-add-version sqlite3

autoantwort committed:
add link to common vcpkg definitions (#17066)

n4z4m3 committed:
Update port version

There was a change in master which updated the port version causing a conflict with this branch so I re-based and updated my port version to 5.

n4z4m3 committed:
Merge branch 'New-Static-Library-Dependencies' of https://github.com/nazame/vcpkg into New-Static-Library-Dependencies

nazame commented:
Are all these changes because I didn't do the force push after re-basing?

vicroms commented:
Probably because I've been merging PRs to master that affected `gdal` as well.

JackBoosY commented:
Wow, the PR contains too many changes, can you split them?

nazame commented:
> Wow, the PR contains too many changes, can you split them?

@JackBoosY I agree it contains too many changes. The problem came because there were conflicts and I thought the way to resolve them would be to re-base the branch with master. However, I didn't to the force push after rebasing which caused all these changes to be pulled in from master.
Is there a way to undo this besides creating a new PR?

JackBoosY commented:
@nazame  You can revert to an earlier commit I think.

ytbigvinny27 reviewed (approved):
🤚🏾

nazame commented:
I wish to close this pull request since the latest update to the gdal package resolves part of it (the geos dependencies) and the other part (the proj shell32.lib and ole32.lib dependencies) are not really necessary. If you us the Additional Dependencies option in MS Visual Studio they are added already and this is the default for a new project in Visual Studio anyway.
